### PR TITLE
Misc build related fixes

### PR DESCRIPTION
**Pull request by kugel-**
Needed to compile against Geany's linkage-cleanup branch but they
are good changes regardless.

### CODE REVIEW

File: codenav/src/utils.c [hunk 1]
```diff
@@ -19,7 +19,7 @@
  *  along with this program; if not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "utils.h"
```
Comment (b4n) [inline codenav/src/utils.c]: don't remove this, it's the include for the plugin's utils.h, aka this file's header

File: codenav/src/utils.c [hunk 2]
```diff
@@ -19,7 +19,7 @@
...
+#include "geanyplugin.h"
```
Comment (b4n) [inline codenav/src/utils.c]: hum, why would we need this if we didn't before?

File: devhelp/src/Makefile.am [hunk 3]
```diff
@@ -25,6 +25,7 @@ devhelp_la_CFLAGS = \
 
 devhelp_la_LIBADD = \
 	$(DEVHELP_LIBS) \
+	$(COMMONLIBS) \
```
Comment (b4n) [inline devhelp/src/Makefile.am]: OK

**Comment by b4n** [comment]
OK, technically most are good, but why are they needed (nut the `COMMONLIBS` thing, which I understand)

File: codenav/src/utils.c (hunk 2 above)
Comment (kugel-) [inline codenav/src/utils.c]: I got this warning, and the change fixed it for me. I didn't realize there is also a local utils.h
  CC       utils.lo
utils.c: In Funktion »compare_strings«:
utils.c:92:2: Warnung: Implizite Deklaration der Funktion »utils_str_equal« [-Wimplicit-function-declaration]
  return (gint)(!utils_str_equal(a, b));

It's bad that that codenav has a header called util.h, it'll always clash. The convention should have been to #include <geany/geanyplugin.h> but that's a lot of changes.

File: geanyminiscript/src/gms_gui.c [hunk 4]
```diff
@@ -26,7 +26,7 @@
  * \brief it is the graphical user interface of the geany miniscript plugin
  */
 #include    "config.h"
-#include    "geany.h"
+#include    "geanyplugin.h"
```
Comment (kugel-) [inline geanyminiscript/src/gms_gui.c]: This change fixes
  CC       gms_gui.lo
In file included from gms_gui.c:57:0:
gms.h:30:8: error: unknown type name 'GeanyFunctions'
 extern GeanyFunctions   *geany_functions;

+#include "geanyfunctions.h" would also work but geanyplugin.h is recommended anyway

File: shiftcolumn/src/shiftcolumn.c [hunk 5]
```diff
@@ -23,20 +23,12 @@
...
 #endif
 
-#include "ui_utils.h"
-
-#include "document.h"
-#include "keybindings.h"
-#include "plugindata.h"
-#include "geanyfunctions.h"
-
 #include <glib.h>
```
Comment (kugel-) [inline shiftcolumn/src/shiftcolumn.c]: This change fixes

  CC       shiftcolumn.lo
shiftcolumn.c: In function 'shift_left_cb':
shiftcolumn.c:95:4: warning: implicit declaration of function 'sci_has_selection' [-Wimplicit-function-declaration]
    if (sci_has_selection(sci)){
    ^
#include "sciwrappers.h" would also work but geanyplugin.h is recommended anyway

File: codenav/src/utils.c [hunk 6]
```diff
@@ -19,6 +19,8 @@
  *  along with this program; if not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <geanyplugin.h>
+
 #include "utils.h"
```
Comment (kugel-) [inline codenav/src/utils.c]: As an alternative one could replace the inappropriate use of utils_str_equal with g_strcmp/0 but i dont like to potentially change the behavior now

**Comment by frlan** [comment]
I'm planning to merge the changes only if referenced Geany changes are getting merged.

**Comment by kugel-** [comment]
These changes don't depend on the Geany changes. They should be applied either way

File: devhelp/src/Makefile.am (hunk 3 above)
Comment (b4n) [inline devhelp/src/Makefile.am]: I now fixed this part in 2d1dee0d9a135f2d6d29c4eead6c6e13e5d10dcd (and the markdown one in 856b3d9a0d065866c8117b031fe82087543b3f6f)

File: geanyminiscript/src/gms_gui.c [hunk 7]
```diff
@@ -26,7 +26,7 @@
...
```
Comment (b4n) [inline geanyminiscript/src/gms_gui.c]: ideally should use `<>`s

File: shiftcolumn/src/shiftcolumn.c [hunk 8]
```diff
@@ -23,20 +23,12 @@
 # include "config.h"
 #endif
 
-#include "geany.h"
-#include "support.h"
+#include "geanyplugin.h"
```
Comment (b4n) [inline shiftcolumn/src/shiftcolumn.c]: `<>`s here too

**Comment by b4n** [comment]
OK, so now I ignored this one when I committed 2d1dee0d9a135f2d6d29c4eead6c6e13e5d10dcd and 856b3d9a0d065866c8117b031fe82087543b3f6f, this "only" "fixes" plugins "incorrectly" including something else than `<geanyplugin.h>`.  But do we have a real guideline on that?  I agree we should and plugins probably should only include `<geanyplugin.h>`, but [the docs](http://www.geany.org/manual/reference/howto.html#helloworld) don't seem to be really clear on this.

And if we do that, maybe there are more plugins to fix? :)

File: codenav/src/utils.c (hunk 2 above)
Comment (LarsGit223) [reply codenav/src/utils.c]: I am not getting that warning if I compile ```codenav/src/utils.c``` from current master.

**Comment by LarsGit223** [comment]
I fixed all usages of ```#include "geanyplugin.h"``` in PR #882.

File: shiftcolumn/src/shiftcolumn.c (hunk 5 above)
Comment (LarsGit223) [reply shiftcolumn/src/shiftcolumn.c]: This seems to have been fixed separately in f2ab51090839b2a93ec8cc3ed0e1e4874ff9b3cf.

File: geanyminiscript/src/gms_gui.c (hunk 4 above)
Comment (LarsGit223) [reply geanyminiscript/src/gms_gui.c]: I do not get that error if I build from current master.

**Comment by LarsGit223** [comment]
IMHO we could close this unmerged if PR #882 gets merged because all issues then seem to have been fixed in other PRs.

@b4n, @kugel-: what do you think?

**Comment by kugel-** [comment]
Makes sense, feel free.

**Comment by LarsGit223** [comment]
Closing in favor of all changes done separately in other PRs.

**Pull request closed without merging**